src/oxli/hashtable.cc

```diff
@@ -482,9 +478,6 @@ uint64_t * Hashtable::abundance_distribution(
                 BoundedCounterType n = get_count(kmer);
                 dist[n]++;
             }
-
-            name.clear();
-            seq.clear();
```

Comment:
`KmerHashIteratorPtr` has a pointer to `seq` which we invalidate by calling `clear()`. The standard apparently doesn't require that `clear` actually goes and sets all characters to `\0` so all it does is set the first one to `\0`. Not sure why we are calling `clear()` in the first place. If at all it should be just after the `while` loop. I think we never noticed before because we don't "subtract" the first character to compute the hash of the second kmer. This is new in the rolling hash ... was pretty puzzled as to why sometimes "AAAAAAAA" would result in a hash that changes as you iterate through the 4-mers in that sequence ... :)

Reply:
Yeah, I can't see any reason why this should be here.